ci/upload-artifacts-template.yml

```diff
@@ -25,13 +33,22 @@ steps:
 
       # Write all include patterns to a file used by rsync.
       echo "${{ join('\n', parameters.includePatterns) }}" > "$BUILD_ROOT/include_patterns.txt"
+      echo "${{ join('\n', parameters.unconditionalIncludePatterns) }}" > "$BUILD_ROOT/unconditional_include_patterns.txt"
```

Comment:
Looks like this join did something unexpected and goofy. It created a single string with literally `\n` in the middle of the two parts, like this:

```
2022-07-21T01:10:48.8960353Z Files matching these patterns will always be included in the binary
2022-07-21T01:10:48.8961119Z build artifact for this job:
2022-07-21T01:10:48.8961665Z vvvvvvvvvvvvvvvvvv
2022-07-21T01:10:48.8962234Z /hw/top_earlgrey/rom.mmi\n/hw/top_earlgrey/otp.mmi
2022-07-21T01:10:48.8962739Z ^^^^^^^^^^^^^^^^^^
```

Reply:
I guess in between `${{` and `}}` is some non-bash preprocessor language, and the details for strings are much less clear to me. Apparently `'\n'` doesn't do what we think, and most likely the first line is bogus too ;)

Reply:
Good catch! I had a slightly different take. I think the `\n`-joining expression is doing its job, but bash's `echo` does not interpret backslash escapes by default! I'll try again with the `-e` flag.

```
$ echo 'foo\nbar'
foo\nbar
$ echo -e 'foo\nbar'
foo
bar
```

Reply:
Ah, good point!

It would be true that whatever processes the join doesn't put in a newline, but I forgot an actual line feed substitution there would not work well for the bash script, heh. So it's the job of `echo` :)

Reply:
Just as some additional context, the `${{` and `}}` indicates an [Azure template variable](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#template-expression-syntax). Azure has three types of variables that evaluate at different points during the pipeline parsing and execution. Template variables get evaluated at "compile-time" which is to say when the YAML file is parsed by Azure. The inner `join(...)` is an [Azure expression](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#join).

Reply:
Just checked the CI results and it seems to have worked!

https://dev.azure.com/lowrisc/opentitan/_build/results?buildId=87486&view=logs&j=12d53c77-8779-5458-10a0-4e65f4237072&t=23fdf818-c6c5-5d82-571c-df0a8b144c7b&l=13

```
Files matching these patterns will be included in the binary build
artifact for this job unless they were downloaded from an upstream job:
vvvvvvvvvvvvvvvvvv
/hw/***
^^^^^^^^^^^^^^^^^^

Files matching these patterns will always be included in the binary
build artifact for this job:
vvvvvvvvvvvvvvvvvv
/hw/top_earlgrey/rom.mmi
/hw/top_earlgrey/otp.mmi
^^^^^^^^^^^^^^^^^^
```

This also neatly resolved the issue of the MMI files not matching the pattern. I believe I'm seeing paths to the MMI files printed out by the `tar` command due to the `-v` flag.